vra/resource_blueprint.go

```diff
@@ -0,0 +1,290 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceBlueprintCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("starting to create vra_blueprint resource")
+	apiClient := m.(*Client).apiClient
+
+	blueprintSpecification := models.Blueprint{
+		Content:         d.Get("content").(string),
+		Name:            d.Get("name").(string),
+		ProjectID:       d.Get("project_id").(string),
+		RequestScopeOrg: d.Get("request_scope_org").(bool),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		blueprintSpecification.Description = v.(string)
+	}
+
+	resp, err := apiClient.Blueprint.CreateBlueprintUsingPOST1(blueprint.NewCreateBlueprintUsingPOST1Params().WithBlueprint(&blueprintSpecification))
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(resp.GetPayload().ID)
+	log.Printf("finished to create vra_blueprint resource with name %s", d.Get("name"))
+
+	return resourceBlueprintRead(d, m)
+}
+
+func resourceBlueprintRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("Reading the vra_blueprint resource with name %s", d.Get("name"))
+	apiClient := m.(*Client).apiClient
+
+	id := d.Id()
+	bpUuid := strfmt.UUID(id)
+
+	resp, err := apiClient.Blueprint.GetBlueprintUsingGET1(blueprint.NewGetBlueprintUsingGET1Params().WithBlueprintID(bpUuid))
+
+	if err != nil {
+		switch err.(type) {
+		case *blueprint.GetBlueprintUsingGET1NotFound:
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	blueprint := *resp.Payload
+	d.Set("content", blueprint.Content)
+	d.Set("content_source_id", blueprint.ContentSourceID)
+	d.Set("content_source_pat", blueprint.ContentSourcePath)
+	d.Set("content_source_sync_at", blueprint.ContentSourceSyncAt)
+	d.Set("content_source_sync_messages", blueprint.ContentSourceSyncMessages)
+	d.Set("content_source_sync_status", blueprint.ContentSourceSyncStatus)
+	d.Set("content_source_type", blueprint.ContentSourceType)
+	d.Set("created_at", blueprint.CreatedAt)
+	d.Set("created_by", blueprint.CreatedBy)
+	d.Set("description", blueprint.Description)
+	d.Set("name", blueprint.Name)
+	d.Set("org_id", blueprint.OrgID)
+	d.Set("project_id", blueprint.ProjectID)
+	d.Set("project_name", blueprint.ProjectName)
+	d.Set("request_scope_org", blueprint.RequestScopeOrg)
+	d.Set("self_link", blueprint.SelfLink)
+	d.Set("status", blueprint.Status)
+	d.Set("total_released_versions", blueprint.TotalReleasedVersions)
+	d.Set("total_versions", blueprint.TotalVersions)
+	d.Set("updated_at", blueprint.UpdatedAt)
+	d.Set("updated_by", blueprint.UpdatedBy)
+	d.Set("valid", blueprint.Valid)
+
+	if err := d.Set("validation_messages", flattenValidationMessages(blueprint.ValidationMessages)); err != nil {
+		return fmt.Errorf("error setting validation_messages in blueprint - error: %#v", err)
+	}
+
+	log.Printf("finished reading the vra_blueprint resource with name %s", d.Get("name"))
```

Comment:
Please review the printf's in this file. If keeping them, perhaps capitalize "finished"?

Reply:
Done.